_posts/2021-06-01-migration-v1-v2.md

```diff
@@ -0,0 +1,130 @@
+
+**Pre-requisites**: having running Apicurio Registry instances for both the version 1.3.x server (exporting from) and the new 2.x server (importing to).
+
+1. Export all the data from Apicurio Registry 1.3.x using the `exportV1` tool. This will generate a `registry-export.zip` file in your current directory.
+
+```
+java -jar apicurio-registry-utils-exportV1-2.0.0-SNAPSHOT-runner.jar http://old-registry.my-company.com/api
+```
+
+2 - Import the zip file to Apicurio Registry 2.x using the import API. Extended documentation on this API can be found [here](https://www.apicur.io/registry/docs/apicurio-registry/2.0.0.Final/getting-started/assembly-managing-registry-artifacts-api.html#exporting-importing-using-rest-api)
+
+```
+curl -X POST "http://new-registry.my-company.com/apis/registry/v2/admin/import" \
+  -H "Accept: application/json" -H "Content-Type: application/zip" \
+  --data-binary @registry-export.zip
+```
+
+3 - Check that all the artifacts are now imported to the new 2.x registry. Run these two commands and compare the count field.
+
+```
+curl "http://old-registry.my-company.com/api/search/artifacts"
+```
+
+```
+curl "http://new-registry.my-company.com/apis/registry/v2/search/artifacts"
+```
+
+
+### Application Migration
+
+If you are using the Apicurio Registry SerDes libraries, you need to change the maven dependencies you are using, as we have repackaged these classes.
+
+In Apicurio Registry 1.3.x the SerDes libraries were provided all with just one maven dependency:
+```
+<dependency>
+    <groupId>io.apicurio</groupId>
+    <artifactId>apicurio-registry-utils-serde</artifactId>
+    <version>1.3.2.Final</version>
+</dependency>
+```
+
+In Apicurio Registry 2.x this has been changed, and the SerDes libraries have been split into three maven dependencies, one for each supported data format: `avro`, `protobuf` and `json schema`.
+
+```
+<dependency>
+    <groupId>io.apicurio</groupId>
+    <artifactId>apicurio-registry-serdes-avro-serde</artifactId>
+    <version>2.0.0.Final</version>
+</dependency>
+<dependency>
+    <groupId>io.apicurio</groupId>
+    <artifactId>apicurio-registry-serdes-protobuf-serde</artifactId>
+    <version>2.0.0.Final</version>
+</dependency>
+<dependency>
+    <groupId>io.apicurio</groupId>
+    <artifactId>apicurio-registry-serdes-jsonschema-serde</artifactId>
+    <version>2.0.0.Final</version>
+</dependency>
+```
+
+You have to take into account this changes in the maven dependencies, and update your applications accordingly.
+
+The refactored SerDes libraries also include some changes in the configuration properties, you can find more information [here](https://www.apicur.io/registry/docs/apicurio-registry/2.0.0.Final/getting-started/assembly-using-kafka-client-serdes.html) , and you can find plenty of examples using Apicurio Registry 2.0.0.Final SerDes libraries [here](https://github.com/Apicurio/apicurio-registry-examples/tree/2.0.x)
```

Comment:
"...in the configuration properties. You can find more details in the [Apicurio Registry user documentation] (...)"